lib/commons/integrity/check/unique_csv_identifiers.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+require 'csv'
+
+module Commons
+  module Integrity
+    class Check
+      # Check that any values in the given columns are unique within the file
+      #
+      # == Configuration Options
+      # * column_names: the columns containing the IDs (default: ["wikidata", "ms_fb"])
+      # * column_case: "fixed" if the column_name must be exactly as specified (default: "any")
+      class UniqueCSVIdentifiers < Base
+        # @return [Array<Error>]
+        # Errors will be in the category `:unique_csv_identifiers`
+        def errors
+          return [] unless pathname_regex.match? pathname.to_s
+          problematic_values.map do |column_name, value|
+            error(unique_csv_identifiers: "Duplicated #{column_name} value: #{value}")
+          end
+        end
+
+        private
+
+        DEFAULT_COLUMN_NAMES = %w[wikidata ms_fb].freeze
+        DEFAULT_COLUMN_CASE = 'any'
+
+        def csv
+          @csv ||= CSV.read(pathname, headers: true)
+        end
+
+        def pathname_regex
+          Regexp.new(my_config.to_h['pathname_regex'] || '.*')
+        end
+
+        def headers
+          csv.headers
+        end
+
+        def column_names
+          (my_config.to_h['column_names'] || DEFAULT_COLUMN_NAMES).map do |column_name|
+            column_name.send(comparison_conversion)
+          end
+        end
+
+        def comparison_type
+          my_config.to_h['column_case'] || DEFAULT_COLUMN_CASE
+        end
+
+        def comparison_conversion
+          return :to_s if comparison_type == 'fixed'
+          :downcase
+        end
+
+        def wikidata_columns
+          headers.select do |header|
+            column_names.include? header.send(comparison_conversion)
+          end
+        end
+
+        def wikidata_columns?
+          !wikidata_columns.all?(&:nil?)
+        end
+
+        def column_values(column_name)
+          csv.map { |row| row[column_name] }
```

Comment:
This is already built into the csv library. There are a few different ways you can get at it, but I suspect the clearest would be via `csv.values_at(column_name)`. That would enable you to inline this, and thus not have two distinct methods which each take a `column_name` parameter.

If you _do_ still end up with two methods like that, then I'd much more inclined to look at whether you'd be better off with a `Column` class, which could then be asked for `.duplicates` etc.  (I'd be tempted to do that anyway, even if there's only a single method taking the column as a parameter, as I suspect it'd make the implementation both simpler and clearer, but it would become even more valuable if there are more than one.)

Reply:
I didn't know about `csv.values_at`, so have changed it to `csv.values_at(column_name).map(&:first)`, as `csv.values_at` wasn't quite a drop in replacement (returning a single-element array per row, instead the value per row).